app/src/main/java/com/babylon/wallet/android/domain/usecases/p2plink/SyncAccountsWithConnectorExtensionUseCase.kt

```diff
@@ -0,0 +1,119 @@
+@file:Suppress("LongParameterList")
+
+package com.babylon.wallet.android.domain.usecases.p2plink
+
+import com.babylon.wallet.android.data.dapp.PeerdroidClient
+import com.babylon.wallet.android.data.dapp.model.Account
+import com.babylon.wallet.android.data.dapp.model.ConnectorExtensionExchangeInteraction
+import com.babylon.wallet.android.data.repository.p2plink.P2PLinksRepository
+import com.babylon.wallet.android.di.coroutines.IoDispatcher
+import com.radixdlt.sargon.extensions.hash
+import com.radixdlt.sargon.extensions.hex
+import com.radixdlt.sargon.extensions.hexToBagOfBytes
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.firstOrNull
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.withContext
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import rdx.works.core.decodeHex
+import rdx.works.core.hash
+import rdx.works.core.preferences.PreferencesManager
+import rdx.works.core.toHexString
+import rdx.works.peerdroid.data.PeerdroidConnector
+import rdx.works.peerdroid.domain.PeerConnectionStatus
+import rdx.works.profile.data.model.apppreferences.P2PLinkPurpose
+import rdx.works.profile.domain.GetProfileUseCase
+import rdx.works.profile.domain.activeAccountsOnCurrentNetwork
+import timber.log.Timber
+import javax.inject.Inject
+
+class SyncAccountsWithConnectorExtensionUseCase @Inject constructor(
+    private val p2pLinksRepository: P2PLinksRepository,
+    private val getProfileUseCase: GetProfileUseCase,
+    private val peerdroidConnector: PeerdroidConnector,
+    private val peerdroidClient: PeerdroidClient,
```

Comment:
⚠️  PeerdroidClient and PeerdroidConnector in the same class? That's not what I had in my mind when implementing those two classes. 😄 
I see what you are trying to achieve below, but isn't the `hasAtLeastOneConnection` from `PeerdroidClient` enough?

Reply:
one more thing is that `PeerdroidClient` is already injected in `MainViewModel` where the `SyncAccountsWithConnectorExtensionUseCase` is used. So I would recommend that this usecase returns a message and then the viewmodel sends it. What do you think?

Reply:
Unfortunately, it's not enough to rely on hasAtLeastOneConnection. I've seen that PeerdroidClient depends on PeerdroidConnector, so I understand what you mean. What about getting rid of PeerdroidClient here and relying only on PeerdroidConnector?

Reply:
To complete my comment on why it's not enough to rely on `hasAtLeastOneConnection`: we have to send the account list message to all the open connections that have a "general" purpose. `hasAtLeastOneConnection` emits `true` every time a connection is established, but we don't have the `connectionId` to be able to check its purpose.

Reply:
As for having `PeerdroidClient` in `MainViewModel` we should think more about it. If we consider PeerdroidClient as a data service component (which I think it is, as it serves as a bridge to the connector extension API - should it be seen as part of the data gathering layering, as a sibling to gateway API / room database / shared preferences or maybe I'm wrong here?), then maybe moving it out from the view model to a repository or a use case would be a good idea? Although it's out of the scope of this PR, but it would result in a cleaner view model.